src/operators/throttleTime.ts

```diff
@@ -85,30 +85,38 @@ class ThrottleTimeSubscriber<T> extends Subscriber<T> {
   }
 
   protected _next(value: T) {
-    if (this.throttled) {
-      if (this.trailing) {
-        this._trailingValue = value;
-        this._hasTrailingValue = true;
-      }
-    } else {
-      this.add(this.throttled = this.scheduler.schedule(dispatchNext, this.duration, { subscriber: this }));
-      if (this.leading) {
-        this.destination.next(value);
-      }
+    if (!this.throttled && this.leading) {
+      this.throttle();
+      this.destination.next(value);
+    } else if (this.trailing) {
+      this._trailingValue = value;
+      this._hasTrailingValue = true;
+    }
+
+    if (!this.throttled) {
+      this.throttle();
```

Comment:
Can we refactor this a little to be:


```ts
if (!this.trottled) {
  this.throttle();
  if (this.leading) {
    this.destination.next(value);
  }
}

if (this.trailing) {
  this._trailingValue = value;
  this._hasTrailingValue = true;
}
```

I'm not sure in it's current configuration it will work as desired, because of that `else if`. If it's trailing, we always want to record the value, not only if we've already started throttling.

Reply:
Then `throttleTime` would emit the same input twice on `{leading:true, trailing: true}`. I guess in the end it doesn't matter when throttleTime is used in conjunction with `distinctUntilChanged`, but it feels wrong regardless.

```typescript
Observable.of(1).throttleTime(100, {leading:true, trailing: true}).subscribe((n) => console.log(n));

// 1
// 1
```

vs. lodash
```typescript
fn = _.throttle((n) => console.log(n), 100, {leading:true, trailing: true});
fn(1);

// 1
```